certbot-apache/certbot_apache/tests/parsernode_test.py

```diff
@@ -41,38 +54,57 @@ class DummyBlockNode(interfaces.BlockNode):
     dirty = False
     enabled = True
     name = ""
+    filename = ""
 
     def save(self, msg):  # pragma: no cover
+        """Save"""
         pass
 
     def add_child_block(self, name, parameters=None, position=None):  # pragma: no cover
+        """Add child block"""
         pass
 
     def add_child_directive(self, name, parameters=None, position=None):  # pragma: no cover
+        """Add child directive"""
         pass
 
     def add_child_comment(self, comment="", position=None):  # pragma: no cover
+        """Add child comment"""
         pass
 
     def find_blocks(self, name, exclude=True):  # pragma: no cover
+        """Find blocks"""
         pass
 
     def find_directives(self, name, exclude=True):  # pragma: no cover
+        """Find directives"""
         pass
 
     def find_comments(self, comment, exact=False):  # pragma: no cover
+        """Find comments"""
         pass
 
     def delete_child(self, child):  # pragma: no cover
+        """Delete child"""
+        pass
+
+    def metadata(self, key):  # pragma: no cover
+        """Metadata"""
         pass
 
     def set_parameters(self, parameters):  # pragma: no cover
+        """Set parameters"""
         pass
 
     def unsaved_files(self):  # pragma: no cover
+        """Unsaved files"""
         pass
 
 
+interfaces.CommentNode.register(DummyCommentNode)
+interfaces.DirectiveNode.register(DummyDirectiveNode)
+interfaces.BlockNode.register(DummyBlockNode)
```

Comment:
For the classes in this file I think we should either:

1. Define a `DummyParserNode` and have a setup similar to what I described at https://github.com/certbot/certbot/pull/7330#discussion_r314443836 in this file as well. This is my preference as I would have complained about the dummy duplicated methods in this file if I would have noticed.
2. Delete these class to `register` and have each class inherit directly from the interface it implements as is currently in `apache-parser-v2`. I don't think separately inheriting from `interfaces.ParserNode` in each of these classes here does anything significant.

Reply:
I went with the first option as I feel it's a cleaner one.